simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,382 @@
+
+def getTruncatedNormal(mean, sd, lower, upper):
+    """
+    Return truncated normal distribution.
+    """
+    return scipy.stats.truncnorm(
+        (lower - mean) / sd, (upper - mean) / sd, loc = mean, scale = sd)
+
+class Person():
+    """
+    Person class for the dataset.
+    """
+
+    def __init__(self):
+        self.index = None
+        self.globalLikes = None
+        self.lived = None
+        self.localLikes = None
+
+class DataGen():
+    """
+    Data generator class for simple-acquaintances example.
+    """
+
+    def __init__(self, numberOfPeople, numberOfPlaces, numberOfGlobal, numberOfLocal,
+            placesLivedMean, placesLivedSD, localLikesVariance, targetSplit,
+            seed, outputDir):
+        self.globalThings = [i for i in range(numberOfGlobal)]
+        self.localLikesVariance = localLikesVariance
+        self.localThings = [i for i in range(numberOfLocal)]
+        self.numberOfPeople = numberOfPeople
+        self.outputDir = outputDir
+        self.places = [i for i in range(numberOfPlaces)]
+        self.placesLivedMean = placesLivedMean
+        self.placesLivedSD = placesLivedSD
+        self.placesLivedUpper = numberOfPlaces
+        self.seed = seed
+        self.rng = self.__createRNG(seed)
+        self.targetSplit = targetSplit
+
+        self.placesAffectingLocalThings = {}
+        self.__generatePALT()  # Fill in placesAffectingLocalThings.
+
+
+    def __createRNG(self, seed):
+        """
+        Return an instance of a random number generator for this class.
+        """
+        return numpy.random.RandomState(seed)
+
+    def __generatePALT(self):
+        """
+        Generate porbabilities of places affecting the likeablity of local things.
+
+        eg: [Place1:[LThing1, Lthing2], Place2:[LThing1,LThing2]]
+        """
+        for i in self.places:
+            self.placesAffectingLocalThings[i] = self.rng.uniform(0, 1, len(self.localThings))
+
+    def __getLikeability(self, person):
+        """
+        Return likeability of localThings based on the places that a person has lived in.
+        """
+        likeability = {}
+        for thing in self.localThings:
+            likeability[thing] = 0
+            for place in person.lived:
+                truncnorm_gen = getTruncatedNormal(self.placesAffectingLocalThings[place][thing], self.localLikesVariance, 0, 1)
+                likeability[thing] += truncnorm_gen.rvs()
+            likeability[thing] /= len(person.lived)
+
+        return likeability
+
+    def __flipCoin(self, bias = 0.5):
+        """
+        Return 1 or 0 based on a biased coin toss. Default bias = 0.5.
+        """
+        return 1 if self.rng.random_sample() < bias else 0
+
+    def __calculateSimilarity(self, p1, p2):
+        """
+        Return similarity measure between two persons (cosine similarity [0.0...1.0]).
+        """
+        person1 = numpy.array(p1.lived 
+                            + list(p1.globalLikes.values()) 
```

Comment:
For breaking up long lines, just use two indentation levels (don't try to align).